### PR TITLE
update msg to be more friendly in farms page

### PR DESCRIPTION
**Pull request by AlaaElattar**
### Description
- Update error msg when there's problem getting user farms.

### Changes

Error msg updated to `Failed to get farms. Please check your connection.`
### Related Issues

https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1599
### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

**Comment by MohamedElmdary** [comment]
I think we should use `normalizeError` method and that string should be the fallback error
because now if any error happen user won't know @AlaaElattar

**Comment by AlaaElattar** [comment]
> I think we should use `normalizeError` method and that string should be the fallback error because now if any error happen user won't know @AlaaElattar

yes, u're right.

**Comment by AlaaElattar** [comment]
> I think we should use `normalizeError` method and that string should be the fallback error because now if any error happen user won't know @AlaaElattar

should be fixed in [29e54d8](https://github.com/threefoldtech/tfgrid-sdk-ts/pull/1703/commits/29e54d86cac98953e826b725853fd8d06b55f83e)

**Review by amiraabouhadid — CHANGES_REQUESTED** [review]
I got this error instead 
Screenshot from 2023-12-17 14-06-49
Screenshot from 2023-12-17 14-06-09

**Comment by AlaaElattar** [comment]
> I got this error msg instead Screenshot from 2023-12-17 14-06-49 Screenshot from 2023-12-17 14-06-09

Fixed. will always show friendly error msg.

**Review by zaelgohary — COMMENTED** [review]
Are we certain that a connection error is the only potential issue? What if there are problems with the Grid Proxy itself? Shouldn't the error message be more inclusive or at least address other possible scenarios beyond network issues?

**Comment by AlaaElattar** [comment]
> Are we certain that a connection error is the only potential issue? What if there are problems with the Grid Proxy itself? Shouldn't the error message be more inclusive or at least address other possible scenarios beyond network issues?


- Here i tried if i can't reach proxy and i got same logs. Do u have an idea how to differentiate the connection error from any other error ? 

image

**Comment by 0oM4R** [comment]
> > Are we certain that a connection error is the only potential issue? What if there are problems with the Grid Proxy itself? Shouldn't the error message be more inclusive or at least address other possible scenarios beyond network issues?
> 
> * Here i tried if i can't reach proxy and i got same logs. Do u have an idea how to differentiate the connection error from any other error ?
> 
> image

@AlaaElattar 
you can check on the error type, all errors should be an instance of a class on types package

example:

- import the type you want to check on here i will import validationError

then check on the catch error
` if (error instanceof ValidationError)`

**Comment by amiraabouhadid** [comment]
found this issue while testing #1733

File: packages/playground/src/dashboard/components/user_farms.vue [hunk 1]
```diff
@@ -235,7 +236,11 @@ export default {
         farmsCount.value = count || filteredFarms.length;
       } catch (error) {
         console.log(error);
-        createCustomToast("Failed to get user farms!", ToastType.danger);
+        if (error instanceof ConnectionError) {
+          createCustomToast("Failed to get farms. Please check your connection.", ToastType.danger);
+        } else {
+          createCustomToast("Failed to get farms.", ToastType.danger);
```
Comment (amiraabouhadid) [inline packages/playground/src/dashboard/components/user_farms.vue:242]: i got this . can you please add "Please check your connection. " here too?
image

Comment (AlaaElattar) [reply packages/playground/src/dashboard/components/user_farms.vue:242]: If i added same msg , what is the point then of checking on error type ?

Comment (MohamedElmdary) [reply packages/playground/src/dashboard/components/user_farms.vue:242]: This issue should be fixed here https://github.com/threefoldtech/tfgrid-sdk-ts/pull/1735 @AlaaElattar @amiraabouhadid

Comment (zaelgohary) [reply packages/playground/src/dashboard/components/user_farms.vue:242]: @amiraabouhadid Another error type not connection related might happen in the else, so I think it's better this way.

**Review by zaelgohary — DISMISSED** [review]

File: packages/playground/src/dashboard/components/user_farms.vue (hunk 1 above)
Comment (zaelgohary) [reply packages/playground/src/dashboard/components/user_farms.vue:242]: I checked #1735 and it indeed fixes the issue in case no connection is needed.

**Comment by AlaaElattar** [comment]
- I removed the connection check on error as when this pr https://github.com/threefoldtech/tfgrid-sdk-ts/pull/1735 merged, this will happen with any case of internet connection.

**Pull request closed without merging**